pkg/storage/replica_raftstorage.go

```diff
@@ -814,15 +821,7 @@ func (r *Replica) applySnapshot(
 	stats.commit = timeutil.Now()
 
 	r.mu.Lock()
-	// We set the persisted last index to the last applied index. This is
-	// not a correctness issue, but means that we may have just transferred
-	// some entries we're about to re-request from the leader and overwrite.
-	// However, raft.MultiNode currently expects this behaviour, and the
-	// performance implications are not likely to be drastic. If our
-	// feelings about this ever change, we can add a LastIndex field to
-	// raftpb.SnapshotMetadata.
-	r.mu.lastIndex = s.RaftAppliedIndex
-	r.mu.lastTerm = 0
+	r.mu.lastIndex, r.mu.lastTerm = lastIndexAndTermInSnapshot(inSnap)
```

Comment:
couldn't you just return `struct lastIndexAndTerm` from `applySnapshot` and save the other call to `lastIndexAndTermInSnapshot`? Nit, but I think it might be clearer.